src/org/ods/component/ScanWithAquaStage.groovy

```diff
@@ -0,0 +1,122 @@
+package org.ods.component
+
+import groovy.transform.TypeChecked
+import groovy.transform.TypeCheckingMode
+import org.ods.services.AquaService
+import org.ods.services.BitbucketService
+import org.ods.services.OpenShiftService
+import org.ods.util.ILogger
+
+@TypeChecked
+class ScanWithAquaStage extends Stage {
+
+    public final String STAGE_NAME = 'Aqua Security Scan'
+    private final AquaService aqua
+    private final BitbucketService bitbucket
+    private final OpenShiftService openShift
+    private final ScanWithAquaOptions options
+
+    @SuppressWarnings('ParameterCount')
+    @TypeChecked(TypeCheckingMode.SKIP)
+    ScanWithAquaStage(def script, IContext context, Map config, AquaService aqua, BitbucketService bitbucket,
+                      OpenShiftService openShift, ILogger logger) {
+        super(script, context, logger)
+        if (!config.resourceName) {
+            config.resourceName = context.componentId
+        }
+        this.options = new ScanWithAquaOptions(config)
+        this.aqua = aqua
+        this.bitbucket = bitbucket
+        this.openShift = openShift
+    }
+
+    protected run() {
+        Map connectionData = openShift.getConfigMapData(context.cdProject, "aqua")
+        // base URL of Aqua server.
+        String aquaUrl = connectionData['aqua.url']
```

Comment:
If I read this right, the name of the key is `aqua.url` - wouldn't `url` be enough as the scope is already the `aqua` `ConfigMap`?

Reply:
changed it also for _registry_ and _secretName_